src/main/java/net/wasdev/wlp/ant/AbstractTask.java

```diff
@@ -427,6 +425,80 @@ protected String findStringInFile(String regexp, File fileToSearch) throws Excep
         return matches;
     }
 
+    /**
+     * Searches the given file for the given regular expression.
+     *
+     * @param regexp
+     *            a regular expression (or just a text snippet) to search for
+     * @param fileToSearch
+     *            the file to search
+     * @return List of Strings which match the pattern. No match results in an
+     *         empty list.
+     */
+    protected List<String> findStringsInFileCommon(String regexp,
+                                                   boolean stopOnFirst, int searchLimit, File fileToSearch) {
+        return findStringsInFileCommon(regexp, stopOnFirst, searchLimit, fileToSearch, Project.MSG_INFO);
+    }
+
+    /**
+     * Searches the given file for the given regular expression, and returns the number of times it occurs.
+     * 
+     * @param regexp
+     *            a regular expression (or just a text snippet) to search for
+     * @param fileToSearch
+     *            the file to search
+     * @return the number of occurrences of the regular expression
+     */
+    public int countStringOccurrencesInFile(String regexp, File fileToSearch) {
+        List<String> lines = findStringsInFileCommon(regexp, false, -1, fileToSearch, Project.MSG_VERBOSE);
+
+        if (lines == null) {
+            return 0;
+        }
+        return lines.size();
+    }
+
+    /**
+     * Wait until the number of occurrences of the regular expression in the file increases
+     * above the given number of previous occurrences.
+     *
+     * @param regexp
+     *            a regular expression to search for
+     * @param timeout
+     *            a timeout, in milliseconds
+     * @param outputFile
+     *            file to check
+     * @param the previous number of occurrences of the regular expression
+     * @return updated line that matched the regexp
+     */
+    public String waitForUpdatedStringInLog(String regexp, long timeout,
+                                     File outputFile, int previousOccurrences) {
+        long waited = 0;
+        final long waitIncrement = 500;
+
+        log(MessageFormat.format(messages.getString("info.search.string"), regexp, outputFile.getAbsolutePath(), timeout / 1000));
+
+        while (waited <= timeout) {
+            List<String> matches = findStringsInFileCommon(regexp, false, -1, outputFile, Project.MSG_VERBOSE);
+            if (matches == null || matches.size() <= previousOccurrences) {
```

Comment:
I think checking the previous timestamp would be more robust than the number of previous occurrences.

Reply:
Thanks for the suggestion.  Since this works for now but could be improved, I've opened a separate issue to address this. https://github.com/WASdev/ci.ant/issues/103